react/DeviceCoordinates.tsx

```diff
@@ -84,13 +84,21 @@ const DeviceCoordinates: React.FC<Props> = ({ onSuccess }) => {
   }
 
   useEffect(() => {
-    navigator.permissions
-      .query({ name: 'geolocation' })
-      .then((result: PermissionStatus) => {
-        if (result.state === 'denied') {
-          setGeolocationPermission('denied')
-        }
-      })
+    try {
+      navigator.permissions
+        .query({ name: 'geolocation' })
+        .then((result: PermissionStatus) => {
+          if (result.state === 'denied') {
+            setGeolocationPermission('denied')
+          }
+        })
+    } catch (e) {
+      // ignoring TypeError handles browsers that don't support `permission`
+      // https://developer.mozilla.org/en-US/docs/Web/API/Navigator/permissions#Browser_compatibility
+      if (!(e instanceof TypeError)) {
+        throw e
```

Comment:
is it ok to rethrow like that?

Reply:
If with "rethrow like that" you mean like "catching only to throw again", this is not a problem in JS. Unlike other languages where the stack trace would be modified, in JS it will not. It would be just as good without the `try catch`.

But I think that it's not a good practice to implement flow control inside a catch block. It's made for exception handlings, not flow control. I think you should check if the `navigator` has `permissions` then execute the methods, not intentionally generate an error (in some cases).

Let me know if I'm being clear and if you agree.

Reply:
>  [...] It would be just as good without the try catch.

Cool!

> But I think that it's not a good practice to use flow control inside a catch block [...]

Agreed, I just don't know exactly how to test for that...

Reply:
ah, nvm, I can just check if it's undefined. I was thinking that I would need to know which browser the user is using, but that does not seem to be the case